comp/dogstatsd/server/server_test.go

```diff
@@ -40,24 +40,26 @@ type serverDeps struct {
 	Log    log.Component
 }
 
-// getAvailableUDPPort requests a random port number and makes sure it is available
-func getAvailableUDPPort() (int, error) {
+// createUDPConnRandomPort starts listening on an random UDP port.
+// If the "close" boolean is set to false, the connection is returned and the
+// caller should taker care of closing it properly if no error is returned.
+func createUDPConnRandomPort(close bool) (*net.PacketConn, int, error) {
```

Comment:
`PacketConn` is an interface, so it should not be necessary to return it by pointer:

```suggestion
func createUDPConnRandomPort(close bool) (net.PacketConn, int, error) {
```

Reply:
You're right I'll address this.
Unfortunately my fix is not helping with the flakiness as not being able to use `udp://127.0.0.1:0` makes it difficult to be 100% sure we've access to this port, I'm still looking for something better than current PR.